build-system/tasks/runtime-test.js

```diff
@@ -541,96 +546,123 @@ function runTests() {
   const sectionMarker =
       (argv.saucelabs || argv.saucelabs_lite) ? 'saucelabs' : 'local';
 
-  let resolver;
-  const deferred = new Promise(resolverIn => {resolver = resolverIn;});
-  new Karma(c, function(exitCode) {
-    if (shouldCollapseSummary) {
-      console./* OK*/log('travis_fold:end:console_errors_' + sectionMarker);
-    }
-    server.emit('kill');
-    if (exitCode) {
-      log(
-          red('ERROR:'),
-          yellow('Karma test failed with exit code ' + exitCode));
+  if (argv.saucelabs || argv.saucelabs_lite) {
+    let batch = 1;
+    const batchSize = 4; // Number of Sauce Lab browsers
+    let left = 0;
+    let right = batchSize;
+    while (left < right) {
+      const configBatch = c;
```

Comment:
What does this line do?

Reply:
Loops through the browser list in batches. I changed "left" and "right" to "startIndex" and "endIndex" for clarity.

Reply:
Creating a new `const` here might not do what you think it does :)

The "constant-ness" of `configBatch` only means that you cannot change what object is being pointed to by `configBatch` (e.g., you can't do `configBatch = null` afterwards inside this block). It doesn't create a copy - so any changes to `configBatch` will also change `c`:

```javascript
~$ node
> const a = {1: 'hello', 2: 'world'}
undefined
> const b = a;
undefined
> b[2] = 'kitty!'
'kitty!'
> b
{ '1': 'hello', '2': 'kitty!' }
> a
{ '1': 'hello', '2': 'kitty!' }
> 
```

Reply:
Makes sense :) Updated to copy the config instead